sasmodels/models/cylinder.c

```diff
@@ -7,50 +7,94 @@ form_volume(double radius, double length)
 }
 
 static double
-fq(double qab, double qc, double radius, double length)
+_fq(double qab, double qc, double radius, double length)
 {
     return sas_2J1x_x(qab*radius) * sas_sinx_x(qc*0.5*length);
 }
 
 static double
-orient_avg_1D(double q, double radius, double length)
+radius_from_excluded_volume(double radius, double length)
+{
+    return 0.5*cbrt(0.75*radius*(2.0*radius*length + (radius + length)*(M_PI*radius + length)));
+}
+
+static double
+radius_from_volume(double radius, double length)
+{
+    return cbrt(M_PI*radius*radius*length/M_4PI_3);
+}
+
+static double
+radius_from_diagonal(double radius, double length)
+{
+    return sqrt(radius*radius + 0.25*length*length);
+}
+
+static double
+effective_radius(int mode, double radius, double length)
```

Comment:
Can `mode` be an enum here (and in similar `switch` constructs elsewhere) so the caller can use symbolic names for the calculation mode rather than magic numbers?

Reply:
This function is never called directly, so an enum would not help.  There is a list of strings associated with each model that turn into a drop box in the gui which returns an integer which is sent to the function.